src/readiness/index.js

```diff
@@ -55,6 +58,11 @@ function GateContext() {
       setTimeout(() => gate.emit(Events.SDK_READY_FROM_CACHE), 0);
     });
 
+    splits.on(Events.SDK_SPLITS_KILL, () => {
+      // Make it async
+      setTimeout(() => gate.emit(Events.SDK_UPDATE), 0);
```

Comment:
We would need to ensure this won't fire if the SDK was not ready before by checking that status is SDK_FIRE_UPDATE

Reply:
removed. I am using SDK_SPLITS_ARRIVED as we talked.